controllers/common.go

```diff
@@ -83,28 +83,70 @@ func IsPaused(annotations map[string]string) bool {
 }
 
 func getNodeSelector(deploymentName string, instance *searchv1alpha1.Search) map[string]string {
-	deploymentConfig := getDeploymentConfig(deploymentName, instance)
-	if deploymentConfig.NodeSelector != nil {
-		return deploymentConfig.NodeSelector
+	if instance.Spec.NodeSelector != nil {
+		return instance.Spec.NodeSelector
 	}
 	var result map[string]string
 	return result
 }
 
 func getImagePullPolicy(deploymentName string, instance *searchv1alpha1.Search) corev1.PullPolicy {
-	deploymentConfig := getDeploymentConfig(deploymentName, instance)
-	if deploymentConfig.ImagePullPolicy != "" {
-		return deploymentConfig.ImagePullPolicy
+	if instance.Spec.ImagePullPolicy != "" {
+		return instance.Spec.ImagePullPolicy
 	}
 	// Dev preview option
 	return corev1.PullAlways
 }
 
+func getPostgresVolume(instance *searchv1alpha1.Search) corev1.Volume {
+	storageClass := instance.Spec.DBStorage.StorageClassName
+	if storageClass != "" {
+		pvcName := getPVCName(storageClass)
+		return corev1.Volume{
+			Name: "postgresdb",
+			VolumeSource: corev1.VolumeSource{
+				PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+					ClaimName: pvcName,
+				},
+			},
+		}
+	}
+	return corev1.Volume{
+		Name: "postgresdb",
+		VolumeSource: corev1.VolumeSource{
+			EmptyDir: &corev1.EmptyDirVolumeSource{},
+		},
+	}
+}
+
+func getContainerArgs(deploymentName string, instance *searchv1alpha1.Search) []string {
+	var result []string
+	switch deploymentName {
+	case apiDeploymentName:
+		if instance.Spec.Deployments.QueryAPI.Arguments != nil {
+			return instance.Spec.Deployments.QueryAPI.Arguments
+		}
+	case collectorDeploymentName:
+		if instance.Spec.Deployments.Collector.Arguments != nil {
+			return instance.Spec.Deployments.Collector.Arguments
+		}
+	case indexerDeploymentName:
+		if instance.Spec.Deployments.Indexer.Arguments != nil {
+			return instance.Spec.Deployments.Indexer.Arguments
+		}
+	case postgresDeploymentName:
+		if instance.Spec.Deployments.Database.Arguments != nil {
+			return instance.Spec.Deployments.Database.Arguments
+		}
+	}
+	log.V(2).Info("Unknown deployment %s ", deploymentName)
```

Comment:
We want to log this only when none of the case conditions match.
```suggestion
	default:
		log.V(2).Info("Unknown deployment %s ", deploymentName)
	}
```

Reply:
Right - but the valid cases would have returned . This  line will only execute for the ones we dont know about.

Reply:
The operators are using this package to log "sigs.k8s.io/controller-runtime/pkg/log" - That does not have the Infof . I see folks using Info . @jlpadilla